sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithTimestamps.java

```diff
@@ -101,7 +101,7 @@ public Duration getAllowedTimestampSkew() {
   public PCollection<T> apply(PCollection<T> input) {
     return input
         .apply("AddTimestamps", ParDo.of(new AddTimestampsDoFn<T>(fn, allowedTimestampSkew)))
-        .setTypeDescriptorInternal(input.getTypeDescriptor());
+        .setCoder(input.getCoder());
```

Comment:
Where is this necessary? In general, having the output type descriptor line up with the input type descriptor should cause the coder inference mechanism to do exactly this.